sonar-db/src/main/java/org/sonar/db/property/InternalPropertyDao.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.property;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.api.utils.System2;
+import org.sonar.db.Dao;
+import org.sonar.db.DbSession;
+
+public class InternalPropertyDao implements Dao {
+
+  private final System2 system2;
+
+  public InternalPropertyDao(System2 system2) {
+    this.system2 = system2;
+  }
+
+  public void save(DbSession dbSession, String key, @Nullable String value) {
```

Comment:
As we discussed recently with @julienlancelot , why not use a mapper here?

Since there is no streaming, there is apparently no specific reason. MyBatis update statement can return the number of updated rows. The mapper method just need to return an `int`.

If we start writing down the full JDBC code, we should discuss where to put the line between using a mapper and not use it.